src/lib.rs

```diff
@@ -11,26 +11,28 @@ pub mod register;
 mod socket;
 pub mod tcp;
 pub mod udp;
-mod uninitialized_device;
+pub mod uninitialized_device;
```

Comment:
Why make this public? Is there a need? The other mods for `tcp` and `udp` are public because we directly access them, but that's not really the intent for `uninitialized_device`, where we re-export the `UninitializedDevice` type at the root.

Reply:
You also have the `InitializeError` which, currently, is not exported anywhere.

Reply:
Maybe we should just export the `InitializeError` instead then? If we make the mod pub, we have to maintain that API going forward in the future, which can be annoying.

Reply:
I'm more worried about the non-existing DHCP implementation that's in the API than the module export :sweat_smile: 
I've update lib.rs to reflect this change.

Reply:
Yeah, DHCP is less relevant now that you can use w5500 with smoltcp, but eh.